dataclients/kubernetes/ingress.go

```diff
@@ -291,7 +322,12 @@ func applyAnnotationPredicates(m PathMode, r *eskip.Route, annotation string) er
 		}
 	}
 
-	r.Predicates = append(r.Predicates, predicates...)
+	for _, p := range predicates {
```

Comment:
nitpicking: would it be nicer to first append into a tmp var, and then use ReplacePredicates?

Reply:
Handled differently :)